File: rb/features.rb
# JellyBellyWiki SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module JellyBellyWikiFeatures
  def self.make_feature(name)
    case name
    when "base"
      JellyBellyWikiBaseFeature.new
    when "ratelimit"
      JellyBellyWikiRatelimitFeature.new
    when "retry"
      JellyBellyWikiRetryFeature.new
    when "test"
      JellyBellyWikiTestFeature.new
    when "timeout"
      JellyBellyWikiTimeoutFeature.new
    else
      JellyBellyWikiBaseFeature.new
    end
  end
end

File: py/features.py
# JellyBellyWiki SDK feature factory

from feature.base_feature import JellyBellyWikiBaseFeature
from feature.test_feature import JellyBellyWikiTestFeature


def _make_feature(name):
    features = {
        "base": lambda: JellyBellyWikiBaseFeature(),
        "test": lambda: JellyBellyWikiTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
